treeherder/client/thclient/client.py

```diff
@@ -552,6 +552,25 @@ def validate(self):
         for d in self.data:
             d.validate()
 
+    def get_chunks(self, chunk_size):
+        """
+        Return a list of new collections broken into chunks of size ``chunk_size``.
+
+        Each chunk will be a ``TreeherderCollection`` of the same
+        type as the original with a max of ``chunk_size`` count of
+        ``TreeherderData`` objects.
+
+        Each collection must then be POSTed individually.
+        """
+        artifact_collections = []
+        data_chunks = [self.data[o:o + chunk_size] for o
```

Comment:
Unfortunately, if the length of items doesn't evenly fall into the chunks, it will drop the last incomplete chunk.  ``izip_longest`` looked promising, but it will fill with empty values that cause problems later down the code path. 

I *could* fill with ``None`` and then handle that possibility in ``get_data_collection`` but that would be inefficient since the for loop would have to loop through all the ``None`` values at the end.

Seems like my original approach may still be the best?